core/src/main/scala/org/apache/spark/status/api/v1/StagesResource.scala

```diff
@@ -143,7 +148,10 @@ private[v1] class StagesResource extends BaseAppResource {
     @Context uriInfo: UriInfo):
   HashMap[String, Object] = {
     withUI { ui =>
-      val uriQueryParameters = uriInfo.getQueryParameters(true)
+      // Decode URI twice here to avoid percent-encoding twice on the query string
+      val decodeURI = URLDecoder.decode(uriInfo.getRequestUri.getRawQuery, UTF_8.name())
+      val uriQueryParameters = new ImmutableMultivaluedMap[String, String](
+        UriComponent.decodeQuery(decodeURI, true))
```

Comment:
Not sure if decoding twice is appropriate here.
If do not change here, the stage page can be opened, but the task sort column button cannot be clicked (locality level etc.)